core/vm/analysis.go

```diff
@@ -19,6 +19,45 @@ package vm
 import (
 )
 
+// fill in segment of operation information array for a block
+func analyzeBlock(ctx *callCtx, pc uint64) (*BlockInfo, error) {
+	info := NewBlockInfo(ctx.contract, pc)
+	code := ctx.contract.Code
+	jumpTable := ctx.interpreter.jt
+	for ; pc < uint64(len(code)); pc++ {
+		op := OpCode(code[pc])
+		operation := jumpTable[op]
+		if operation == nil {
+			return nil, &ErrInvalidOpCode{opcode: op}
+		}
+		info.minStack += operation.minStack
+		info.maxStack += operation.maxStack
+		info.constantGas += operation.constantGas
+		
+		if op >= PUSH1 && op <= PUSH32 {
+
+			// decode push data to PushInfo
```

Comment:
Is this a TODO? I can see it is used in `opPush1`, but this is where `PushInfo` is supposed to be created, right?

Reply:
Yes, a TODO.   I couldn't understand the original logic for dealing with the "virtual zeros" at the end of the bytecode, which also looked to be inefficient.  What I did in aleth was to pad the bytecode with some real zero bytes.  So I need to discuss this with @holiman.

Reply:
Also, the PushInfo is used in all of the push operations.